### ISSUE TITLE
dyndns result matching

### ISSUE THREAD

**Issue by fliiiix** [opnsense/plugins#555]
**TL;DR**

It would be helpful if DDNS on verbose would somehow accurately print why the result matching failed.

**Detailed description**

I debugged today a case where the DDNS Result Match didn't work. So I enabled verbose login and I think the logging is not enough:

here is the error message in question:
https://github.com/opnsense/plugins/blob/c6fbdd45fc4f1c2f5da804d735e1855bc6afdd27/dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc#L1117

This produces error messages like this in the log:

```
opnsense: /services_dyndns_edit.php: Dynamic DNS: (Error) Result did not match.
```

It would be helpful to add maybe a log entry what was expected and what it actually was:

So I added something like this:
```
$status = "Dynamic DNS: (Error) Result did not match. ({$this->_dnsResultMatch} != {$data})";
```

which produces that something like this:

```
opnsense: /services_dyndns_edit.php: Dynamic DNS: (Error) Result did not match. (good XXX.248.XX.XX != good XXX.248.XX.XX)
```

This could be enough but in my case there was a extra space between good and the IP so I only saw that when I started to print the strings as array with `print_r`

```
opnsense: /services_dyndns_edit.php: arr m: Array ( [0] => g [1] => o [2] => o [3] => d [4] => [5] => [6] => X [7] => X [8] => X [9] => . [10] => 2 [11] => 4 [12] => 8 [13] => . [14] => X [15] => X [16] => . [17] => X [18] => X )
```

**Comment by fichtner** [comment]
Contributor missing, sorry.
